tmt/result.py

```diff
@@ -132,6 +132,7 @@ class Result(BaseResult):
     starttime: Optional[str] = None
     endtime: Optional[str] = None
     duration: Optional[str] = None
+    data_path: Optional[str] = None
```

Comment:
Declare the field as `Path` rather than `str`, it's a path after all. See `serialize` and `unserialize` callbacks e.g. in `log` field for conversion between `Path` and the on-disk format which would be `str`.

Also, lease update docs describing result format, `specs/plan/result.fmf` (I hope I got the path right).

Reply:
Ack. WRT serialization, possible `None` needs to be taken care of. As long as the field is `Optional`, it may store `None`, therefore this would be better:

```python
serialize=lambda path: None if path is None else str(path),
unserialize=lambda value: None if value is None else Path(value)
```